### PR TITLE
Added an option for formatting  response in the admin api

### PR DESCRIPTION
**Pull request by phawazzzy**
**### Added an option for formatting response in the admin API**.
This is a fix for the issue [#73](https://github.com/TryGhost/Ghost-SDK/issues/73
)
The admin api will have a default formatResponse of true if not added in the options, like so 
`const api = new GhostAdminAPI({
    url: 'http://localhost:2368',
    key: 'admin apikey',
    // key: '80da03aa41787cf5589eafc048',
    version: "v3",
    formatResponse: true
  });`

Then by just checking the value of the formatResponse, if true the formatting would happen, if not it wouldn't it..

The formatting  happens in the makeResourceRequest function for resource types that requires uploading a file(images or theme), and in the makeUploadRequest function for resourceType that requires files
@simplytunde

### CODE REVIEW

**Comment by dabsclement** [comment]
this fix presents options for formatResponse

File: packages/admin-api/lib/index.js [hunk 1]
```diff
@@ -301,7 +305,7 @@ module.exports = function GhostAdminAPI(options) {
     //the resource type can be a post, page, an image, a theme,
     //the function is taking in three parameters, which are the resourcetype(the type of what you want to upload), data(the payload i.e file), endpoint(the route you want to upload to)
     function makeUploadRequest(resourceType, data, endpoint) {
-        //the headers which is the content type with a boundary
+        //the headers which is the content type with a boundaryUnauthorized
```
Comment (nextwebb) [inline packages/admin-api/lib/index.js:308]: //the headers which is the content type with a boundary	
this should be in a comment

File: packages/admin-api/lib/index.js [hunk 2]
```diff
@@ -314,13 +318,16 @@ module.exports = function GhostAdminAPI(options) {
...
             //return a promise data
             .then((data) => {
-                //checking if the data resourcetype is an array, if it isnt an array return the data[resourceType]
-                if (!Array.isArray(data[resourceType])) {
-                    return data[resourceType];
-                }
-
-                if (data[resourceType].length === 1 && !data.meta) {
-                    return data[resourceType][0];
+                if (config.formatResponse === true) {
```
Comment (nextwebb) [inline packages/admin-api/lib/index.js:321]: seems okay to me; returning "res.data" if formatResponse is set to false

File: packages/admin-api/lib/index.js [hunk 3]
```diff
@@ -338,24 +345,27 @@ module.exports = function GhostAdminAPI(options) {
...
+                    return data[resourceType];
+                }
+                // return data[resourceType[0] index of data if data[resourceType] length equals 1 ans data.meta is undefined
+                if (data[resourceType].length === 1 && !data.meta) {
+                    return data[resourceType][0];
+                }
+                //copy the values of all of the enumerable own properties from data.meta as meta data[resourceType]. Returns data[resourceType]
+                return Object.assign(data[resourceType], {meta: data.meta});
+            } else {
+                return Object.assign(data, {meta: data.meta});
```
Comment (nextwebb) [inline packages/admin-api/lib/index.js:360]: well i guess we cant cater for all datatype cases, at least we know the user wants unformatted response ,      " return Object.assign(data, {meta: data.meta}); " seems fine.

File: packages/content-api/lib/index.js [hunk 4]
```diff
@@ -86,13 +86,17 @@ export default function GhostContentAPI({url, host, ghostPath = 'ghost', version
...
+            if (formatResponse) {
+                if (!Array.isArray(res.data[resourceType])) {
+                    return res.data[resourceType];
+                }
+                if (res.data[resourceType].length === 1 && !res.data.meta) {
+                    return res.data[resourceType][0];
+                }
+                return Object.assign(res.data[resourceType], {meta: res.data.meta});
+            } else {
+                return res.data;
```
Comment (nextwebb) [inline packages/content-api/lib/index.js:98]: seems okay returning "res.data " in this case for an unformatted response.
just a quick question,
 why didn't you add the meta data to the response ??

Comment (phawazzzy) [reply packages/content-api/lib/index.js:98]: It wasn't necessary... But it can be added

File: packages/admin-api/lib/index.js (hunk 1 above)
Comment (phawazzzy) [reply packages/admin-api/lib/index.js:308]: it is a comment

Comment (nextwebb) [reply packages/admin-api/lib/index.js:308]: alright

**Pull request merged**